ingestion/src/metadata/ingestion/models/patch_request.py

```diff
@@ -375,10 +393,20 @@ def build_patch(
 
         # For a user editable fields like descriptions, tags we only want to support "add" operation in patch
         # we will remove the other operations.
+        # This will only be applicable if the override_metadata field is set to False.
         if restrict_update_fields:
             updated_operations = JsonPatchUpdater.from_restrict_update_fields(
                 restrict_update_fields
-            ).update(patch)
+            ).update(patch, override_metadata=override_metadata)
+
+            # if the only operation is to update the sourceHash, we'll skip the patch
+            # since this will only happen when we filter out the REMOVE and REPLACE ops
+            # and if the sourceHash is updated in this case further updates will not be processed
+            if (
+                len(updated_operations) == 1
+                and updated_operations[0]["path"] == "/sourceHash"
```

Comment:
should you do `updated_operations[0].get("path")` so worst case scenario it does not fail. I am not 100% of the specs of `updated_operations` feel free to let me know if it is ok as is

Reply:
I think it should be ok. Still I'll update it to get to be on safer side